File: krishideep_app/lib/models/crop.dart
class Crop {
  final String id;
  final String name;
  final String nameHindi;
  final String description;
  final double expectedYield; // kg per acre
  final double profitMargin; // percentage
  final double sustainabilityScore; // 0-10 scale
  final List<String> suitableConditions;
  final String imageUrl;

  Crop({
    required this.id,
    required this.name,
    required this.nameHindi,
    required this.description,
    required this.expectedYield,
    required this.profitMargin,
    required this.sustainabilityScore,
    required this.suitableConditions,
    this.imageUrl = '',
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'nameHindi': nameHindi,
      'description': description,
      'expectedYield': expectedYield,
      'profitMargin': profitMargin,
      'sustainabilityScore': sustainabilityScore,
      'suitableConditions': suitableConditions,
      'imageUrl': imageUrl,
    };
  }

  factory Crop.fromJson(Map<String, dynamic> json) {
    return Crop(
      id: json['id'],
      name: json['name'],
      nameHindi: json['nameHindi'],
      description: json['description'],
      expectedYield: json['expectedYield'].toDouble(),
      profitMargin: json['profitMargin'].toDouble(),
      sustainabilityScore: json['sustainabilityScore'].toDouble(),
      suitableConditions: List<String>.from(json['suitableConditions']),
      imageUrl: json['imageUrl'] ?? '',
    );
  }
}

File: krisideepweb/src/models/Crop.ts
export interface Crop {
  id: string;
  name: string;
  nameHindi: string;
  description: string;
  expectedYield: number; // kg per acre
  profitMargin: number; // percentage
  sustainabilityScore: number; // 0-10 scale
  suitableConditions: string[];
  imageUrl: string;
}

export interface CropJson {
  id: string;
  name: string;
  nameHindi: string;
  description: string;
  expectedYield: number;
  profitMargin: number;
  sustainabilityScore: number;
  suitableConditions: string[];
  imageUrl: string;
}

export const cropFromJson = (json: CropJson): Crop => ({
  id: json.id,
  name: json.name,
  nameHindi: json.nameHindi,
  description: json.description,
  expectedYield: json.expectedYield,
  profitMargin: json.profitMargin,
  sustainabilityScore: json.sustainabilityScore,
  suitableConditions: json.suitableConditions,
  imageUrl: json.imageUrl || '',
});

export const cropToJson = (crop: Crop): CropJson => ({
  id: crop.id,
  name: crop.name,
  nameHindi: crop.nameHindi,
  description: crop.description,
  expectedYield: crop.expectedYield,
  profitMargin: crop.profitMargin,
  sustainabilityScore: crop.sustainabilityScore,
  suitableConditions: crop.suitableConditions,
  imageUrl: crop.imageUrl,
});
